src/main/java/uk/gov/companieshouse/filetransferservice/service/impl/AmazonFileTransferImpl.java

```diff
@@ -0,0 +1,301 @@
+            }});
+            return null;
+        }
+    }
+
+    /**
+     * Delete an object in S3
+     *
+     * @throws SdkClientException
+     */
+    @Override
+    public void deleteFile(String fileId) {
+        AmazonS3 s3Client = getAmazonS3Client();
+        validateS3Details(s3Client);
+
+        if (validatePathExists(s3Client)) {
+            s3Client.deleteObject(new DeleteObjectRequest(getAWSBucketName(), fileId));
+        } else {
+            throw new SdkClientException(FOLDER_ERROR_MESSAGE);
+        }
+
+    }
+
+    /**
+     * Get an object in S3
+     *
+     * @throws SdkClientException
+     */
+    protected S3Object getObjectInS3(String location, AmazonS3 s3Client) {
+
+        String bucket = getBucketFromS3Location(location);
+        String fileId = getKeyFromS3Location(location);
+
+        if (!s3Client.doesObjectExist(bucket, fileId))
+            throw new SdkClientException("S3 path does not exist: " + location);
+
+        return s3Client.getObject(new GetObjectRequest(bucket, fileId));
+    }
+
+
+    protected String getBucketFromS3Location(String location) {
+        String path = location.substring(SPLIT_S3_PATH_SUBSTRING);
+        return path.split("/")[0];
+    }
+
+    /**
+     * Get the fileId from the S3 location
+     */
+    protected String getKeyFromS3Location(String location) {
+        String path = location.substring(SPLIT_S3_PATH_SUBSTRING);
+        String[] pathSegments = path.split("/");
+
+        StringBuilder stringBuffer = new StringBuilder();
+        for (int i = 1; i < pathSegments.length - 1; i++) {
+            stringBuffer.append(pathSegments[i]);
+            stringBuffer.append('/');
+        }
+        stringBuffer.append(pathSegments[pathSegments.length - 1]);
+
+        return stringBuffer.toString();
+    }
+
+    /**
+     * Get the bucket name on S3
+     */
+    private String getAWSBucketName() {
+        return getSplitS3path()[0];
+    }
+
+    private String getPathIfExists() {
+        if (getSplitS3path().length > 1) {
+            return getSplitS3path()[1] + SUFFIX;
+        } else {
+            return "";
+        }
+    }
+
+    /**
+     * Get the S3 path split
+     */
+    private String[] getSplitS3path() {
+        String s3Path = configuration.getS3Path();
+        String path = s3Path.substring(SPLIT_S3_PATH_SUBSTRING);
+        return path.split("/", GET_S3_PATH_FROM_SPLIT);
+    }
+
+    /**
+     * Get the proxy host if it has been defined
+     *
+     * @return A {@link String} or null
+     */
+    private String getProxyHost() {
+        return System.getenv(PROXY_HOST_PROPERTY);
+    }
+
+    /**
+     * Get the proxy port if it has been defined
+     *
+     * @return An {@link Integer} or null
+     */
+    private Integer getProxyPort() {
+        String proxyPortString = System.getenv(PROXY_PORT_PROPERTY);
+        return proxyPortString == null ? null : Integer.valueOf(proxyPortString);
+    }
```

Comment:
We shouldn't be using `System.getenv` directly. If we need to access environment variables directly, we should be using the [environment reader](https://github.com/companieshouse/environment-reader-library) library. However, a better was would be to pass them through properties.

Reply:
@HarryBosworth should we use environment reader, properties or both?

Reply:
done using properties